src/expander.ml

```diff
@@ -97,12 +112,54 @@ let expand_var_exn t var syn =
             (String_with_vars.Var.describe var)
         ])
 
+let expand_artifact ~dir ~loc t a s =
+  let path = Path.Build.relative dir s in
+  let name = Path.Build.basename path in
+  let dir = Path.Build.parent_exn path in
+  match a with
+  | Pform.Artifact.Mod kind -> begin
+      match t.lookup_module with
+      | None ->
+        None
+      | Some lookup_module -> begin
+          match lookup_module ~dir (Module.Name.of_string name) with
+          | None ->
+            Some [Value.String (s ^ Pform.Artifact.ext a)]
```

Comment:
Here we return a fictitious filename when the module being referred to does not exist, which will cause an error later (albeit with a misleading error message). Same for libraries below.